### PR TITLE
Monitored Trips - Middleware GraphQL support

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
## Description

Works with https://github.com/ibi-group/otp-middleware/pull/243.

This PR supports OTP-middleware itinerary existence checks using GraphQL. Variables for the corresponding `plan` query are included with the monitored trip data when a trip is saved and the trip existence is checked. A new field `leg.originalMode` is introduced for saving and new/existing trip to revert route mode overrides made by the UI. The PR also updates the UI so that the route numbers are shown properly in the Saved Trips page.

## PR Checklist
- [ ] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [ ] Are all languages supported (Internationalization/Localization)?
- [ ] Are appropriate Typescript types implemented?

### CODE REVIEW

File: lib/actions/user.js [hunk 1]
```diff
@@ -169,14 +174,35 @@ function convertRequestToSearch(config) {
   }
 }
 
+// TODO move to a utils file
+function areRequestsSame(qp1, qp2) {
```
Comment (amy-corson-ibigroup) [inline lib/actions/user.js]: I feel like we've talked about this before, but would it be easier to compare these by excluding unwanted values, stringifying the objects, and comparing the strings?

File: lib/actions/user.js [hunk 2]
```diff
@@ -169,14 +174,35 @@ function convertRequestToSearch(config) {
...
+  // Everything except modes.
```
Comment (amy-corson-ibigroup) [inline lib/actions/user.js]: Also, why do we exclude modes from this check?

**Review by amy-corson-ibigroup — APPROVED** [review]
Looks good! Thank you for cleaning up some of these `routeModeOverride` functions!

File: lib/actions/user.js (hunk 1 above)
Comment (binh-dam-ibigroup) [reply lib/actions/user.js]: Better than stringify, I used deep comparison in https://github.com/opentripplanner/otp-react-redux/pull/1178/commits/ed5f0ad46b92ac710b956ccf09ad6b3d9ebf4054 (We already use a library for that!)

File: lib/actions/user.js (hunk 2 above)
Comment (binh-dam-ibigroup) [reply lib/actions/user.js]: This is for comparing two GraphQL variables from the same trip search, where we send multiple mode combinations. I have clarified the comment in https://github.com/opentripplanner/otp-react-redux/pull/1178/commits/7817c5b9f592a7c4d5d78999d813f28e555d515b.

**Review by josh-willis-arcadis — APPROVED** [review]
LGTM!

**Pull request merged**